lib/pester.rb

```diff
@@ -53,16 +53,28 @@ def self.retry_action(opts = {}, &block)
 
     opts[:max_attempts].times do |attempt_num|
       begin
-        result = yield block
-        return result
+        return yield(block)
       rescue => e
-        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
-        reraise_error = opts[:reraise_error_classes] && opts[:reraise_error_classes].include?(e.class)
-        message_reraise = opts[:message] && !e.message[opts[:message]]
+        retry_error_classes = opts[:retry_error_classes]
+        retry_error_messages = opts[:retry_error_messages]
+        reraise_error_classes = opts[:reraise_error_classes]
+
+        if retry_error_classes
+          if retry_error_messages
+            retry_decision = retry_error_classes.include?(e.class) && retry_error_messages.any? { |m| e.message[m] }
+          else
+            retry_decision = retry_error_classes.include?(e.class)
+          end
+        elsif retry_error_messages
+          retry_decision = retry_error_messages.any? { |m| e.message[m] }
+        elsif reraise_error_classes && reraise_error_classes.include?(e.class)
+          retry_decision = false
+        else
+          retry_decision = true
+        end
```

Comment:
Can we just bust the decision-making code into a different method with its own specs? This would get around the ABC rubocop warning I've been ignoring, because I haven't had a chance to actually, substantively refactor the code to just return a boolean.